orttraining/orttraining/test/training_api/common/synthetic_data_loader.h

```diff
@@ -17,7 +17,14 @@
 
 #include <memory>
 #include <utility>
+#ifdef _WIN32
```

Comment:
> _WIN32 [](http://example.com/codeflow?start=6&length=7)

This should be _MS_VER so it does not affect other windows compilers. However, I suggest to look how we can fix the code first rather than suppressing it.

Reply:
changed to _MSC_VER. The error is in the header [variant](https://en.cppreference.com/w/cpp/header/variant), so can't fix it here

Reply:
There could be no warnings in the system headers. I suspect the we have a bug in the code.